x-pack/test/alerting_api_integration/common/lib/get_event_log.ts

```diff
@@ -31,9 +32,11 @@ export async function getEventLog(params: GetEventLogParams): Promise<IValidated
     throw new Error('no events found yet');
   }
 
-  const events: IValidatedEvent[] = (result.data as IValidatedEvent[]).filter(
-    (event) => event?.event?.provider === provider
-  );
+  // filter events to matching provider and requested actions
+  const events: IValidatedEvent[] = (result.data as IValidatedEvent[])
+    .filter((event) => event?.event?.provider === provider)
+    .filter((event) => event?.event?.action)
+    .filter((event) => actionsSet.has(event?.event?.action!));
   const foundActions = new Set(
     events.map((event) => event?.event?.action).filter((event) => !!event)
```

Comment:
I realise this didn't change in this PR but this looks wrong and makes it harder to understand what this is actually testing for.
```suggestion
    events.map((event) => event?.event?.action).filter((action) => !!action)
```